### PR TITLE
Simplify collection script

### PR DESCRIPTION
**Pull request by bookinabox**
Making this WIP pull request for easier reviewing.

Work remaining:

~~- [ ] Figuring out type of Kata installation~~
- [x] Find missing third party tools
~~- [ ] Determine which problem_patterns can be removed and why~~ (Decided to defer this to a future ticket)

### CODE REVIEW

File: src/runtime/data/kata-collect-data.sh.in [hunk 1]
```diff
@@ -310,16 +310,6 @@ show_runtime_log_details()
 	end_section
 }
 
-show_throttler_log_details()
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh.in:313]: The changes on this commit look good. Apart from that, the commit ebbfa321ae132a02577240ab903d948c4646a267 introduced some handling for throttler that we should remove as well. Could you please try to `git-revert` that?

File: src/runtime/data/kata-collect-data.sh [hunk 2]
```diff
@@ -18,8 +24,8 @@ typeset -r containerd_shim_v2=$(command -v "$containerd_shim_v2_name" 2>/dev/nul
 typeset -r kata_monitor_name="kata-monitor"
 typeset -r kata_monitor=$(command -v "$kata_monitor_name" 2>/dev/null)
 
-typeset -r issue_url="@PROJECT_BUG_URL@"
-typeset -r script_version="@VERSION@ (commit @COMMIT@)"
+typeset -r issue_url="$PROJECT_BUG_URL"
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh:27]: Any reason for not `typeset -r issue_url=https://github.com/kata-containers/kata-containers/issues/new` here?

File: src/runtime/data/kata-collect-data.sh [hunk 3]
```diff
@@ -5,8 +5,14 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+PROJECT_BUG_URL="https://github.com/kata-containers/kata-containers/issues/new"
+
+RUNTIME_VERSION_INFO=$( kata-runtime --version | sed -n 's/^.*: //p' )
+read -r VERSION COMMIT <<< $( echo $RUNTIME_VERSION_INFO )
+PROJECT_NAME="Kata Containers"
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh:12]: `PROJECT_NAME` should be in lower case (not a environment variable) and typeset'ed alongside the others.

File: src/runtime/data/kata-collect-data.sh.in (hunk 1 above)
Comment (bookinabox) [reply src/runtime/data/kata-collect-data.sh.in:313]: I'm confused why the revert needs to occur. Does adding back that parameter do anything? The calls to find_system_journal_problems with the throttler are already removed.

File: src/runtime/data/kata-collect-data.sh (hunk 2 above)
Comment (bookinabox) [reply src/runtime/data/kata-collect-data.sh:27]: No

File: src/runtime/data/kata-collect-data.sh (hunk 3 above)
Comment (bookinabox) [reply src/runtime/data/kata-collect-data.sh:12]: changed

File: src/runtime/data/kata-collect-data.sh.in (hunk 1 above)
Comment (wainersm) [reply src/runtime/data/kata-collect-data.sh.in:313]: you are right @bookinabox , ignore my comment here.

File: src/runtime/data/kata-collect-data.sh [hunk 4]
```diff
@@ -803,6 +802,17 @@ read_osbuilder_file()
 	cat "$file"
 }
 
+show_latest_kata_version()
+{
+	# kata-check will not run network checks as root user, but this script requires root to run
+	local latest_version=$(sudo -u nobody kata-runtime kata-check --check-version-only 2>/dev/null || echo "error")
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh:838]: Maybe a better implementation:
```
local rc=0
local latest_version=$(sudo -u nobody kata-runtime kata-check --check-version-only 2>/dev/null || rc=$?)
if [ "$rc" -ne 0 ]; then ....
```

File: src/runtime/data/kata-collect-data.sh [hunk 5]
```diff
@@ -803,6 +802,17 @@ read_osbuilder_file()
...
+	if [ "$latest_version" = "error" ]; then
+		msg "Unable to check for latest version. Check if system is connected"
+	else
+		msg "$latest_version"
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh:842]: In case of no latest version, it will print something like this:

```
Running `kata-collect-data.sh` version `2.5.0-rc0 (commit 4f49423c91285fce6399da034c2867948b4dc616-dirty)` at `2022-08-08.18:42:57.977553756+0000`.
No newer release available
```

What does it print in case of a newer release is found?

File: src/runtime/data/kata-collect-data.sh [hunk 6]
```diff
@@ -250,7 +250,6 @@ show_runtime_configs()
 	# add in the standard defaults for good measure "just in case"
 	configs+=" /etc/kata-containers/configuration.toml"
 	configs+=" /usr/share/defaults/kata-containers/configuration.toml"
-	configs+=" /etc/kata-containers/configuration.toml"
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh]: I think this change is not related. As well as the changes in `tools/packaging/kernel/configs/fragments/x86_64/.config`

File: src/runtime/data/kata-collect-data.sh (hunk 5 above)
Comment (bookinabox) [reply src/runtime/data/kata-collect-data.sh:842]: ]"Newer %s release available: %s (url: %v, date: %v)\n", desc, details.version, details.url, details.date)"

If there is a newer release it should call showLatestRelease, which outputs the above formatted string.

File: src/runtime/data/kata-collect-data.sh (hunk 6 above)
Comment (bookinabox) [reply src/runtime/data/kata-collect-data.sh]: the config file is definitely not related, but I removed that line because it is redundant of line 250 above, if I understand the code correctly.

**Comment by bookinabox** [comment]
I believe I marked this as ready only for testing purposes. There still are a few more components that need to be implemented unfortunately.

File: src/runtime/data/kata-collect-data.sh [hunk 7]
```diff
@@ -82,14 +86,17 @@ usage()
...
 
 Description: Run this script as root to obtain a markdown-formatted summary
-  of the environment of the @PROJECT_NAME@ installation. The output of this script
+  of the environment of the $project_name installation. The output of this script
   can be pasted directly into a github issue at the address below:
 
       $issue_url
 
+Commands:
+ show-image-details : outputs the yaml describing the image
```
Comment (jodh-intel) [inline src/runtime/data/kata-collect-data.sh:98]: Nit: capitalise acronyms:

```suggestion
 show-image-details : outputs the YAML describing the image
```

File: src/runtime/data/kata-collect-data.sh [hunk 8]
```diff
@@ -242,10 +249,9 @@ show_runtime_configs()
 	show_quoted_text "" "$configs"
 
 	# add in the standard defaults for good measure "just in case"
-	configs+=" /etc/@PROJECT_TAG@/configuration.toml"
-	configs+=" /usr/share/defaults/@PROJECT_TAG@/configuration.toml"
-	configs+=" @CONFIG_PATH@"
-	configs+=" @SYSCONFIG@"
+	configs+=" /etc/kata-containers/configuration.toml"
+	configs+=" /usr/share/defaults/kata-containers/configuration.toml"
+	configs+=" /etc/kata-containers/configuration.toml"
```
Comment (jodh-intel) [inline src/runtime/data/kata-collect-data.sh:254]: This is fine, but it makes me wonder if `configs` should be a proper array.

File: src/runtime/data/kata-collect-data.sh [hunk 9]
```diff
@@ -805,37 +750,70 @@ show_image_details()
...
+		msg "$details"
 	else
 		msg "No initrd"
 	fi
 
+}
+
+show_initrd_details()
+{
+
```
Comment (jodh-intel) [inline src/runtime/data/kata-collect-data.sh:803]: Nit: Extraneous blank line.

File: src/runtime/data/kata-collect-data.sh [hunk 10]
```diff
@@ -854,6 +832,18 @@ read_osbuilder_file()
 	cat "$file"
 }
 
+show_latest_kata_version()
+{
+	# kata-check will not run network checks as root user, but this script requires root to run
+	local rc=0
+	local latest_version=$(sudo -u nobody kata-runtime kata-check --check-version-only 2>/dev/null || rc=$?)
+	if [ "$rc" -ne 0 ]; then
+		msg "Unable to check for latest version. Check if system is connected"
```
Comment (jodh-intel) [inline src/runtime/data/kata-collect-data.sh:841]: I'd make this message slightly clearer:

```suggestion
		msg "Unable to check for latest version. Check if network is available"
```

File: src/runtime/data/kata-collect-data.sh [hunk 11]
```diff
@@ -854,6 +832,18 @@ read_osbuilder_file()
 	cat "$file"
 }
 
+show_latest_kata_version()
+{
+	# kata-check will not run network checks as root user, but this script requires root to run
+	local rc=0
+	local latest_version=$(sudo -u nobody kata-runtime kata-check --check-version-only 2>/dev/null || rc=$?)
```
Comment (jodh-intel) [inline src/runtime/data/kata-collect-data.sh:839]: This function should probably honour the `KATA_CHECK_NO_NETWORK` env var. If set, this script should not try to connect to the network.

**Review by jodh-intel — COMMENTED** [review]
Thanks @bookinabox - a few comments...

**Comment by bookinabox** [comment]
Messed up a rebase somehow. Will delete this comment after I fix it.

**Comment by wainersm** [comment]
hi @bookinabox ! Please add the `Fixes #xxx` to at least one commit so the commit message check will pass.

File: src/runtime/data/kata-collect-data.sh [hunk 12]
```diff
@@ -821,6 +836,10 @@ show_details()
 
 main()
 {
+	if [ "$1" = "show-image-details" ]; then
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh:911]: Those changes belong to the next commit (https://github.com/kata-containers/kata-containers/pull/4736/commits/0678452283463731de2cdb3ca8256c895c318a86).

File: src/runtime/data/kata-collect-data.sh [hunk 13]
```diff
@@ -89,11 +89,14 @@ Usage: $script_name [options]
 Summary: Collect data about an installation of $project_name.
 
 Description: Run this script as root to obtain a markdown-formatted summary
-  of the environment of the $PROJECT_NAME installation. The output of this script
+  of the environment of the $project_name installation. The output of this script
   can be pasted directly into a github issue at the address below:
 
       $issue_url
 
+Commands:
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh:97]: You will need to adjust the usage header from `Usage: $script_name [options]` to `Usage: $script_name [command] [options]`

File: src/runtime/data/kata-collect-data.sh [hunk 14]
```diff
@@ -844,6 +844,49 @@ show_latest_kata_version()
 	fi
 }
 
+show_third_party_tools()
+{
+	not_found="Tool not found"
+	start_section "Third Party Tools"
+
+	conmon_version="$(conmon --version | awk -v FS="(^conmon version)|\n" '{print $2}' | xargs || true)"
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh:852]: I recommend to check the command exist first, otherwise the output is not so nice:
```
<summary>Third Party Tools</summary>
<p>

./kata-collect-data.sh: line 852: conmon: command not found
conmon version: Tool not found
./kata-collect-data.sh: line 856: crio: command not found
cri-o version: Tool not found
containerd version: 1.6.7
./kata-collect-data.sh: line 865: crictl: command not found
critools version: Tool not found
./kata-collect-data.sh: line 870: gperf: command not found
gperf version: Tool not found
./kata-collect-data.sh: line 874: kubectl: command not found
kubernetes version: Tool not found
libseccomp version: 2.5.1
runc version: 1.1.3

</p>
</details>

```

File: src/runtime/data/kata-collect-data.sh [hunk 15]
```diff
@@ -844,6 +844,49 @@ show_latest_kata_version()
...
+	critools_version="$(crictl --version | awk -v FS="(^crictl version)|\n" '{print $2}' | xargs || true)"
+	[ -z $critools_version ] && critools_version=$not_found
+	msg "critools version: $critools_version"
+
+	# Not in vagrant
+	gperf_version="$(gperf --version | awk -v FS="(GNU gperf)|\n" '{print $2}' | xargs || true)"
+	[ -z $gperf_version ] && gperf_version=$not_found
+	msg "gperf version: $gperf_version"
+
+	kubernetes_version="$(kubectl version --output=json | jq -r '.clientVersion.gitVersion' || true)"
```
Comment (wainersm) [inline src/runtime/data/kata-collect-data.sh:874]: kubectl will find a kubeconfig file on some paths and/or the `KUBECONFIG` variable. So it can fail in case the config file is not found.

File: src/runtime/data/kata-collect-data.sh (hunk 14 above)
Comment (bookinabox) [reply src/runtime/data/kata-collect-data.sh:852]: Yeah. I am checking it, but I forgot to redirect stderr

File: src/runtime/data/kata-collect-data.sh (hunk 15 above)
Comment (bookinabox) [reply src/runtime/data/kata-collect-data.sh:874]: Do you mean that we should do better failure handling, or that there should be a better way to do this?

I'm going to redirect all the standard errors for the comment above.

**Comment by bookinabox** [comment]
Because my internship is ending soon, I'm intending on adding any work on problem_patterns and kata-deployment type into this PR.

I am working on finding the kata-deployment type, at least for kata-deploy, but that will likely go into a different PR.

**Comment by bookinabox** [comment]
Just a rebase^

**Comment by bookinabox** [comment]
Static check was failing with:
 
 ERROR: Expected 'EOF' here anchor, in src/runtime/data/kata-collect-data.sh found: EOT
 
 I replaced all instances of EOT with EOF  (This was not a change I made, but just how the original kata-collect-data.sh.in file worked).
 
It would be  better for me to test everything in vagrant, but I have been recently having troubles with vagrant and I'm not sure if I can resolve them fast enough.

**Review by wainersm — APPROVED** [review]

**Comment by bookinabox** [comment]
:/ Metrics failing still. I know we recently updated metrics for the same FIO tests to be more lenient. Perhaps if we re-ran it again it might pass.
